app/controllers/spree/retail/shopify/hooks/refunds_controller.rb

```diff
@@ -0,0 +1,24 @@
+module Spree
+  module Retail
+    module Shopify
+      module Hooks
+        class RefundsController < HooksController
+          def create
+            shopify_refund = ShopifyAPI::Refund.find(shopify_refund_id, params: { order_id: shopify_order_id })
+            Spree::Retail::Shopify::GenerateRefundOrder.new(shopify_refund).process
```

Comment:
Based on my reading of `GenerateRefundOrder`, there are lots of reasons it _might_ fail. But without specific insight into how we might rationally recover, I think your idea is a good one. Assuming the webhook method of synching a refund fails, can it be imported separately?